packages/sdk/src/streamStateView_Space.ts

```diff
@@ -131,10 +112,20 @@ export class StreamStateView_Space extends StreamStateView_AbstractContent {
         }
     }
 
-    private async decryptSpaceImage(encryptedImage: EncryptedData): Promise<ChunkedMedia> {
-        const keyPhrase = contractAddressFromSpaceId(this.streamId)
-        const plaintext = await decryptDerivedAESGCM(keyPhrase, encryptedImage)
-        return ChunkedMedia.fromBinary(plaintext)
+    public async getSpaceImage(): Promise<ChunkedMedia | undefined> {
+        if (this.encryptedSpaceImage) {
+            const initialEncryptedData = this.encryptedSpaceImage // Store the initial state
+            const spaceAddress = contractAddressFromSpaceId(this.streamId)
+            const context = spaceAddress.toLowerCase()
+            const plaintext = await decryptDerivedAESGCM(context, this.encryptedSpaceImage)
+
+            // Ensure the state has not changed during the async operation
+            if (this.encryptedSpaceImage === initialEncryptedData) {
+                this.spaceImage = ChunkedMedia.fromBinary(plaintext)
+                this.encryptedSpaceImage = undefined
+            }
+        }
```

Comment:
This is almost perfect. However, now we have issues calling this multiple times in a row. Which, as you know, in react - is absolutely something that happens all the time.

There are two issues
1) in the normal case, calling multiple times in a row will kick off multiple calls to decryptDericedAESGM - this isn't terrible, just a big waste of cpu
2) if encryptedSpaceImage changes between calls, the first call will return undefined.

I don't think #2 is going to happen much, but if it does we should return an image instead of undefined (rendering the old image is better than a broken image)

Reply:
you might have seen the in-between commits. There is a guard `initialEncryptedData` that should prevent the first call from overriding newer calls.

Reply:
needed help from chatgpt. Changes:
- if multiple calls to `getSpaceImage()` are made while an decryption is running, then return the same promise if the `this.encryptedSpaceImage` is the same.
- if `this.encryptedSpaceImage` is updated while the decryption function is running, then add to the `decryptionQueue` so that the decryption requests are processed in order.